test/browsercontext-device.spec.ts

```diff
@@ -58,4 +59,76 @@ describe('device', (suite, { browserName }) => {
+          <button>hi</button>
+  `);
+    await new Promise(x => setTimeout(x, 500));
+    await page.evaluate(() => window.scroll(0, 100));
+    expect(await page.evaluate(() => window.scrollY)).toBe(100);
+
+    await page.evaluate(() => window.scroll(0, 200));
+    expect(await page.evaluate(() => window.scrollY)).toBe(200);
+
+    await page.goto(server.PREFIX + '/input/scrollable.html');
+    expect(await page.evaluate(() => window.scrollY)).toBe(0);
+    await context.close();
+  });
+
+  it('should reset scroll top after a navigation', (test, {browserName}) => {
+    test.skip(browserName === 'webkit');
+  }, async ({server, contextFactory, playwright, contextOptions}) => {
+    const device = playwright.devices['iPhone 6'];
+    const context = await contextFactory({
+      ...contextOptions,
+      ...device,
+      viewport: {
+        width: 400,
+        height: 400,
+      }
+    });
+    const page = await context.newPage();
+    await page.goto(server.PREFIX + '/input/scrollable.html');
+    await page.evaluate(() => window.scroll(0, 100));
+
+    await page.goto(server.PREFIX + '/input/scrollable2.html');
+    expect(await page.evaluate(() => window.scrollY)).toBe(0);
+    await context.close();
+  });
+
+  it('should scroll to a precise position with mobile scale', (test, {browserName}) => {
```

Comment:
This test seems very similar to the previous one. Let's combine them?

Reply:
They test distinct issues which are not going to be solved with the same patch, so I'd prefer they remain separate.